### PR TITLE
Leia fault find

### PR DESCRIPTION
**Pull request by matthuisman**

### CODE REVIEW

**Comment by matthuisman** [comment]
@facakente

First build to try:
https://jenkins.kodi.tv/job/peak3d/job/inputstream.adaptive/job/PR-453/1/artifact/cmake/addons/build/zips/inputstream.adaptive+android-armv7/inputstream.adaptive-2.4.5.zip

This is 2.4.5 with half its commits removed.

Please try on 18.7 (and 18.6 if possible)
Not all features may work - but we are just looking for what is causing the crash.

**Also, after you change Kodi versions - it would be best to go into android apps and clear all Kodis app data and cache. So its using fresh settings.**

**Maybe before doing anything else - do the above with 18.7 nightly with its built-in IA,
That may be all that is required**

Peak has told me before that updating IA on android doesn't completely update it.
So there may be conflicts

Your actual issue may be because you have manually updated IA - it doesn't like that.
Even when you swap Kodi versions - if you don't clear data - you will have the new version playing havoc with the built-in version

1) install downloaded zip
2) restart kodi
3) uninstall inputstream adaptive
4) restart kodi
5) install downloaded zip
6) restart kodi
7) test

**Comment by facakente** [comment]
Ok with this build in 18.7 , with DV , only get sound no image . Like problem I experienced in the past, no crash though .

With hdr10, it crashes and kicks me out.

**Comment by facakente** [comment]
With stable leia , same input stream works fine with hdr10, it takes around 3 sec to match 24p frame rate but it plays fine .

DV , only sound no image .


So there is a commit needed to make DV to work.  But there is something else messing with HDR10 and crashing in nightly leia .

**Comment by matthuisman** [comment]
but HDR works fine in Netflix?

Doing new build now with all the 2.4.5 commits reverted (so just 2.4.4)

**Comment by facakente** [comment]
Yes it plays fine , only problem with Netflix is that maximum I get is 1080p, but still DV signal . I remember this issue which was corrected with some commit from IA version .

**Comment by matthuisman** [comment]
@facakente 
Try this build:
https://jenkins.kodi.tv/job/peak3d/job/inputstream.adaptive/job/PR-453/2/artifact/cmake/addons/build/zips/inputstream.adaptive+android-armv7/inputstream.adaptive-2.4.5.zip

It says 2.4.5 - but it's just 2.4.4
Again, just looking for crashing on 18.6 and 18.7

Please try different resolutions etc.
Try L1 enabled but selecting 1080p stream etc etc.
Try different audio channels etc etc.
Basically - try find what is causing the crash

**Comment by facakente** [comment]
Ok this one 18.6, stable leia , Disney hdr10 and dv , only sound no image , no crashing.

Netflix plays perfectly no problems

**Comment by matthuisman** [comment]
is that on 18.7 or 18.6 or both?

**Comment by facakente** [comment]
This one with nightly 18.7 , DV only sound no image , no crash.

Hdr10 sound at beginning , but it stops and gets stuck , no image , playback issues definitely  and crashes eventually it I try to make it play .

Netflix only problem with resolution , plays in 1080p, with hdr though

**Comment by matthuisman** [comment]
Again, im not worried about DV or HDR10.
I'm trying to find what is causing the crash only.

Did the crash happen when using lower resolutions?
If you disable HDR10 - did crash happen?

If you can tell me exactly what is causing the crash, then it will tell us where to look

**Comment by facakente** [comment]
In the nightly crashing only happens if I try to hear something from files when playing hdr10 as there is no sound . No crash with dv or l1 deactivated .

**Comment by matthuisman** [comment]
jenkins bui

**Comment by matthuisman** [comment]
jenkins build this please

**Comment by matthuisman** [comment]
@facakente

Can you try installing Kodi 18.6 from below: 
http://mirrors.kodi.tv/releases/android/arm/kodi-18.6-Leia-armeabi-v7a.apk

then try installing IA 2.4.5 from below: 
http://f.mjh.nz/inputstream.adaptive-2.4.5-android-armv7.zip 

Remember to follow this when installing IA on android: 
https://forum.kodi.tv/showthread.php?pid=2947397#pid2947397

This will tell us if the issue is in Kodi 18.7 or if it's in IA 2.4.5

Again - just looking for the "crash behaviour".

**Comment by facakente** [comment]
@matthuisman no crashing with 18.6 leia (either with hdr10 and DV) and addon works fine with that IA version.

**Comment by facakente** [comment]
i must say that currently with this IA version, both netflix and disney (might need some tweaks) work perfectly, at least for me. Framerate issues i had are gone.

**Comment by matthuisman** [comment]
That's great.
What you mean by "need some tweaks?"

please keep using 18.6 then in meantime - there is actually a pretty big bug in 18.7 that causes stuttering etc on shield - that means a 18.7.1 (or 18.8) will be released pretty soon.

That issue was fixed with this PR:
https://github.com/xbmc/xbmc/pull/17945

I'll see if I can get a build of 18.7 with that commit to see if that also fixes your crashing issue

Otherwise, maybe the crashing is due to switching to JNI from NDK (only big Android change I see between 18.7 & 18.8)
https://github.com/xbmc/xbmc/pull/17613

**Comment by matthuisman** [comment]
here is a new 18.7 nightly with that commit in it (please test)
http://mirrors.kodi.tv/nightlies/android/arm/Leia/kodi-20200525-0442d706-Leia-armeabi-v7a.apk

If that still has issues, then I'll try get a build with this PR merged
https://github.com/xbmc/xbmc/pull/17931

**Comment by facakente** [comment]
still crashing with this build u sent me, movie starts and crashes after a sec. (HDR10 and DV)

I need to disable widevine l1 to not crash.

**Comment by matthuisman** [comment]
thanks.
Can you confirm that movies that don't have disney+ logo work fine?

When testing, please turn off Skip Intros in settings.

**Comment by facakente** [comment]
Yes those with no Disney logo work fine , indeed those who cause the crash have the logo at beginning , example mandalorian or Togo .
No problem with or without intros , it worked fine.

**Comment by matthuisman** [comment]
Can you please try this 18.7 build
http://mirrors.kodi.tv/test-builds/android/arm/kodi-20200521-6e4c8210-revertjnileia-armeabi-v7a.apk

That has that large android commit reverted.
Note: That commit has to stay - but this will give us a clue where the issue is

**Comment by facakente** [comment]
Ok with this one there is no crash , works fine. Didn’t watch much so I cannot tell if there is stuttering , but looked fine.

Definitely no crashing .

**Comment by matthuisman** [comment]
Thank you!

I'll let the right people know and see how we go.
Feel free to use that build or 18.6 in the meantime :)

So just to be sure - everything on that last build works 100%?
Even DV framerate changes?

**Comment by facakente** [comment]
Yes regarding problems I reported everything is working fine , netflix perfect , Disney needs some tweaks , when changing signal to DV , it detects , then it loses , finally it retrieves DV signal , during this period there is a sec with no image only sound , later on image but no sound , finally retrieving audio , minor Issue in my opinion because it plays fine after this.

**Comment by matthuisman** [comment]
only DV?
HDR10+ fine on the same disney content?

Also, confirm Netflix works 100% on all IA versions and all kodi versions (even 18.7)?
I dont think netflix uses multi-periods - so that may be why you dont have issues with it

**Comment by glennguy** [comment]
> during this period there is a sec with no image only sound , later on image but no sound , finally retrieving audio

sounds to be exactly the issue described here: https://github.com/xbmc/xbmc/pull/17713
Should be resolved once this is

**Comment by facakente** [comment]
Also with hdr10, noticed only happens with movies where there are Disney logos , with others it works fine.

**Comment by matthuisman** [comment]
@facakente 

Please try this Matrix build:
https://jenkins.kodi.tv/job/Android-ARM/18239/artifact/kodi-20200527-fc78832e-PR17957-merge-armeabi-v7a.apk

It has a new PR by peak that should fix the crashing issue when secure widevine is enabled.
It will also be backported to Leia

**Comment by facakente** [comment]
can u send me the right version of IA? As the one from leia is not compatible with matrix build.

**Comment by facakente** [comment]
I have installed the one from here https://jenkins.kodi.tv/job/peak3d/job/inputstream.adaptive/job/Matrix/

though there is only sound when playing with kodi matrix build.

No crash though.

**Comment by matthuisman** [comment]
Try this Leia build please. Should already have IA included i n it.

http://mirrors.kodi.tv/test-builds/android/arm/kodi-20200528-fc0a622f-release_leia-armeabi-v7a.apk

**Comment by facakente** [comment]
Ok tested that build , not sure if u were expecting the minor issue with hdr10 and dv where it loses signal and retrieves later on , being fixed , because I can tell you it still the same . With movies with Disney logos it struggles a bit it detects hdr signal , then loses it and retrieves again.

**Comment by matthuisman** [comment]
PLEASE - we arn't trying to fix that issue - I've said this a lot of times.
(thats another issue which has a PR to fix - but that PR isn't ready)

We are trying to fix the CRASHING issue.

**Can you confirm that is not happening?**

**Comment by facakente** [comment]
Yes but already mentioned that was not happening before , the crashing issue since you sent me last build was already tackled .

Last 2 builds before this one , matrix and leia both no crashing . Working fine

**Comment by matthuisman** [comment]
Oh, that last Leia build had a big android chunk removed

"That has that large android commit reverted.
Note: That commit has to stay - but this will give us a clue where the issue is"

This latest one is with the correct fix  that is actually merged in Leia.

**Comment by facakente** [comment]
Ok resuming matrix and leia U sent me there isn’t any crashing , hope this helps

**Comment by matthuisman** [comment]
The PR that fixes your other issue is most likely this:
https://github.com/xbmc/xbmc/pull/17713
(But that currently looks like it won't be merged in it's current form.)

I'll see if I can get a Leia build with that in it so you can confirm

**Comment by matthuisman** [comment]
@facakente 
Ok, here is a Leia build to try
http://mirrors.kodi.tv/test-builds/android/arm/kodi-20200529-963c8773-leia_dmx-armeabi-v7a.apk

That has the fix for crashing issue, and the DMX short period fix as well.

**Comment by facakente** [comment]
Fixed , I can confirm.

**Comment by matthuisman** [comment]
So everything works 100% in that build?

**Comment by facakente** [comment]
Yep , all problems reported with Disney and Netflix , have been all sorted. 

Thanks for your work and help.

**Comment by matthuisman** [comment]
ok, next stage is convincing the right guy to help get a correct fix that won't affect other stuff.
I offered him a back massage...

**Comment by facakente** [comment]
@matthuisman I see some Issues on Disney with this last build you sent me , where image is not fluid , seems it stops . Are we aware of any issue ? U mentioned there was a known issue with shield where it stutters .


Indeed in firestick when playing 1080p movies it has so many frames dropped , whereas hdr10 or DV , it does not happen.

**Comment by facakente** [comment]
@matthuisman any chance to have a look at the above, as far I can see with hdr content it does not stutter/ frame dropping, on the other hand when playing HD content , it stutters a lot as it drops frame regularly.

I have installed latest builds but issue persists:

http://mirrors.kodi.tv/test-builds/android/arm/kodi-20200531-9afd7ab5-Leia-armeabi-v7a.apk


also netflix experiences the same problem, dont know if the problem is the build or recent changes in IA.

Let me know if you are aware of issue.. and there is something to follow up.

**Comment by matthuisman** [comment]
im really not sure.
Have you tried 18.7.1?
http://mirrors.kodi.tv/releases/android/arm64-v8a/kodi-18.7.1-Leia-arm64-v8a.apk

There are so many android commits going on at the moment.
I've spent quite a bit of time trying to get builds with certain things changed etc.
I'm now refocusing on my add-ons.

I think its best to post over here:
https://forum.kodi.tv/forumdisplay.php?fid=164

**Comment by facakente** [comment]
Yes tried last one for firestick , but issue still there.
